Fix AppSettingDict to store FileField values as storage paths instead of pickled files

huubbouma opened the pull request:
This change updates file-backed settings to be persisted via Django’s default_storage. Uploaded files are saved to storage and only the resulting storage path is stored in the setting value. When reading a file setting, the stored path is resolved back into a File object.

The update also handles cleanup of replaced or cleared file values and preserves compatibility with existing stored file settings.

Commit:
Fix AppSettingDict to store FileField values as storage paths instead of pickled files

Commit:
Forgot to commit admin.py changes

ruuddeklerk commented:
Requested changes:

- Move all file related code to files.py

- app_settings should not enforce any storage backend or path on the FileField. These can be configured through FileField.storage and FileField.upload_to and be respected by apps_settings

-  code in __init__.py lines 109-125 can be encapsulated within a single if statement:
`if _is_file_setting(group_name, setting_name):
    files._set_item(setting, new_value)
` 

-  let's discuss _open_as_django_file in person

Commit:
Refactor FileField handling: use field storage/upload_to, avoid open file handles

- Extract file logic into files.py; use field.storage instead of default_storage
- Use field.generate_filename() so upload_to callables are respected
- __getitem__ returns raw storage path string instead of an opened File handle
- set_item only saves on actual UploadedFile, skips no-op on existing value

huubbouma commented:
Fixed the code. Now file related code in files.py
no open files anymore
Manually tested in django project
